Docs: Improvements to release contribution process (#922)
grafana/writers-toolkit: docs/sources/contribute/release-notes/index.md

## What's new documentation development process

What's new content is published to the website through the website content management system (CMS).

Because this platform is meant to be used by the entire organization, by default anyone can contribute and publish to What's new, without the need for approval.
_Quality assurance is a conversation within and between contributing teams and internal stakeholders_, but there are some best practice guidelines described in the last two sections of this topic.

**What does "Published" mean?**

It's important to understand that, in the context of the CMS, the word "published" has a slightly different meaning than in general use:

- **Published**: Your entry is complete and in **Published** status.
  It's either visible on the external _What's new in Cloud_ page or is going to automatically become visible on the release date.

> - **Live**: Your entry is visible on the _What's new in Cloud_ page.

Reviewer: So does Published and Live sometimes mean the same thing?

Author: No, what we're getting at is that published in this context doesn't mean live, though it does in most other contexts.

Reviewer: the wording still leads me to believe that they can _sometimes_ be the same thing. "...visible on the external _What's new in Cloud_ page" and "Your entry is visible on the _What's new in Cloud_ page." are equivalent to me.

Author: Ah okay, I get what you mean. So I actually answered your first question incorrectly. Something can be published and not live as well as published and live. It's dependent on release date. Let me see if I can find a way to make that clearer.

Reviewer: gotcha! Maybe an example would help here.

### What's new timing

Enter release notes into the CMS _two to four weeks before the feature is available_, depending on the size of the product or feature.

For Grafana versioned releases, have your release notes entered in the CMS _by the cut-off date_ communicated by the delivery team.

This gives the Go To Market (GTM) team time for promotion and enablement.
For more information, refer to the [Record Announce Document Ship (RADS) guidelines](https://wiki.grafana-ops.net/w/index.php/Engineering/RADS).

### Create a What's new entry

<!-- vale Grafana.GoogleWill = NO -->

When you're ready to add a What's new entry, complete the following steps:

1. Go to the What's new collection of the [website CMS](https://admin.grafana.com/content-admin/#/collections/whats-new) and click **New Feature** at the top of the page.
1. Fill out [the CMS fields](#cms-fields).
1. Click **Save**.
   The entry is now in **Draft** status and the CMS opens a pull request in the `grafana/website` repository.

> 1. Preview your entry in context.

Author: ```suggestion
1. (Optional) Preview your entry on GitHub. While the CMS provides a preview of your entry, it doesn't render video. Previewing your entry on GitHub allows you to see your entry in the context of the website and view videos.
```

- **Release date in the past** - Your entry will be visible in the website-generated preview.

- **Release date in the future** - Your entry is only visible in the preview of the _internal feed_.
  To see it, in the deploy preview URL, replace `/docs/grafana-cloud` with the heading for your entry.
  For example: `https://deploy-preview-18347-zb444pucvq-uw.a.run.app/whats-new/#create-subtables-in-table-visualizations`.

> 1. If your entry is ready to publish, proceed to the next step.

Reviewer: Does this numbering start over because the bulleted list above is left-aligned?

Author: Yep, good catch. I'll fix that and push to the PR again.

   - To publish from GitHub, merge your PR.
   - To publish from the CMS:

   1. In the **Status** drop-down, click **Ready**.
   1. In the **Publish** drop-down, click **Publish now**.

<!-- vale Grafana.GoogleWill = YES -->

The entry appears in [What's new in Cloud](https://grafana.com/docs/grafana-cloud/whats-new/) on the release date you've entered. If the date is in the past, it appears immediately.

For Grafana versioned releases, the content you enter in the CMS is published in the versioned What's new at a later date.
To understand the process of creating release notes for Grafana versioned releases, refer to [Create the versioned release notes](#create-the-versioned-release-notes).

{{< admonition type="caution">}}
If you add an entry to the CMS after the relevant versioned What's new has already been published, you'll need to open a PR to also add it to the versioned What's new yourself.
{{< /admonition >}}

### CMS fields

<!-- vale Grafana.GoogleWill = NO -->
<!-- prettier-ignore-start -->

| Field | Description | Guidance |
|---|---|---|
| FEATURE NAME | Short headline for the feature. | For example, _Grafana OnCall integration for Alerting_. |

> | FEATURE RELEASE DATE | Date and time in UTC that you want this note to be live. | This should also be the feature release date. If the feature is behind a feature toggle and gets rolled out only to a fraction of users the date is when the feature was first available to users opting in.<br></br>If you've opened a review PR, you must merge it before the date you've added here. If you enter a date that has passed, the website publishes the note on the next build. |

Reviewer: ```suggestion
| FEATURE RELEASE DATE | Date and time in UTC that you want this note to be live. | This should also be the feature release date. If the feature is behind a feature toggle and gets rolled out only to a fraction of users, the date is when the feature was first available to users opting in.<br></br>If you've opened a review PR, you must merge it before the date you've added here. If you enter a date that has passed, the website publishes the note on the next build. |
```

Reviewer 2: We shouldn't use `<br>` tags to introduce line breaks. If we want multiple paragraphs, we should use `<p>` tags: https://developers.google.com/style/tables#multi-paragraph-table-cells

Author: I'll fix that in VS code and push again.

## What's new documentation development process

What's new content is published to the website through the website content management system (CMS).

Because this platform is meant to be used by the entire organization, by default anyone can contribute and publish to What's new, without the need for approval.
_Quality assurance is a conversation within and between contributing teams and internal stakeholders_, but there are some best practice guidelines described in the last two sections of this topic.

**What does "Published" mean?**

It's important to understand that, in the context of the CMS, the word "published" has a slightly different meaning than in general use:

- **Published**: Your entry is complete and in **Published** status.
  It's either visible on the external _What's new in Cloud_ page or is going to automatically become visible on the release date.

> - **Live**: Your entry is visible on the _What's new in Cloud_ page.

Author: @chri2547 - Is this better?

### What's new timing

Enter release notes into the CMS _two to four weeks before the feature is available_, depending on the size of the product or feature.

For Grafana versioned releases, have your release notes entered in the CMS _by the cut-off date_ communicated by the delivery team.

This gives the Go To Market (GTM) team time for promotion and enablement.
For more information, refer to the [Record Announce Document Ship (RADS) guidelines](https://wiki.grafana-ops.net/w/index.php/Engineering/RADS).

### Create a What's new entry

<!-- vale Grafana.GoogleWill = NO -->

When you're ready to add a What's new entry, complete the following steps:

1. Go to the What's new collection of the [website CMS](https://admin.grafana.com/content-admin/#/collections/whats-new) and click **New Feature** at the top of the page.
1. Fill out [the CMS fields](#cms-fields).
1. Click **Save**.
   The entry is now in **Draft** status and the CMS opens a pull request in the `grafana/website` repository.

> 1. (Optional) Preview your entry on GitHub.

Author: @chri2547 - Hopefully this is clearer. Further explanation on line 72.

Reviewer: 👍 , clearer - thanks!

>    While the CMS provides a preview of your entry, it doesn't render video. Previewing your entry on GitHub allows view videos and see your entry in the context of the website.

Author: ```suggestion
   While the CMS provides a preview of your entry, it doesn't render video. Previewing your entry on GitHub lets you view videos and see your entry in the context of the website.
```
